src/plugins/robots/generic/simulator/colored_blob_perspective_camera_default_sensor.cpp

```diff
@@ -77,15 +77,15 @@ namespace argos {
                 * So that the origin is up-left, the i axis goes to the right, and the j axis goes down
                 */
                SInt32 unI =
-                  - m_cCamEntity.GetImagePxWidth() /
+                  static_cast<UInt32>(- m_cCamEntity.GetImagePxWidth() /
                   m_cCamEntity.GetImageMtWidth() *
                   (m_cLEDRelative.GetY() -
-                   m_cCamEntity.GetImageMtWidth() * 0.5f);
+                   m_cCamEntity.GetImageMtWidth() * 0.5f));
                SInt32 unJ =
-                  - m_cCamEntity.GetImagePxHeight() /
+                  static_cast<UInt32>(- m_cCamEntity.GetImagePxHeight() /
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary? Also, this cast is wrong because it loses the sign.

Reply:
This variable should be called `nJ` (not `unJ`) and the cast should be `static_cast<SInt32>`. Also, I think consideration should be given here to whether truncating the value is the correct option (foor, ceil, or round should probably be applied first).